lib/cereal/postgres/sql_schema.go

```diff
@@ -102,17 +102,26 @@ CREATE TABLE cereal_workflow_results (
 -- New and Completed tasks also notify a channel that can be
 -- subscribed to by the workflows for more timely notification of new
 -- task events related to their workflow.
-CREATE TYPE cereal_task_status AS ENUM('success', 'failed');
+CREATE TYPE cereal_task_status AS ENUM('success', 'failed', 'worker_lost');
+
+CREATE TYPE cereal_task_state  AS ENUM('queued', 'running');
 
 CREATE TABLE cereal_tasks (
     id BIGSERIAL PRIMARY KEY,
     workflow_instance_id BIGINT NOT NULL,
-    try_remaining INT NOT NULL DEFAULT 1,
     enqueued_at   TIMESTAMPTZ NOT NULL DEFAULT NOW(),
     updated_at    TIMESTAMPTZ NOT NULL DEFAULT NOW(),
     start_after   TIMESTAMPTZ NOT NULL DEFAULT NOW(),
     task_name     TEXT NOT NULL,
-    parameters    BYTEA
+    parameters    BYTEA,
+    task_state    cereal_task_state NOT NULL DEFAULT 'queued',
+    worker_id     TEXT
+);
+
+CREATE TABLE cereal_busy_task_workers (
```

Comment:
can a worker forget about a task and then still have a claim to it because of this table?